mongodb/intro/date-aggregation/aggregation-with-date-expression-operators-v.md

```diff
@@ -0,0 +1,127 @@
+---
+author: Stefan-Stojanovic
+
+levels:
+  - beginner
+  
+aspects:
+  - introduction
+
+type: normal
+
+category: must-know
+
+links:
+
+---
+# Aggregation With All Date Operators
+---
+## Content
+
+Now that we've seen all the date operators and know how to use them, let's take a look at an example of all operators being used in the same aggregation.
+
+**Note:** We are using the same document[1] as in the previous insights.
+
+Example with all operators:
+```js
+db.pokemon.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      dayOfTheYear: { $dayOfYear: "$Date" },
+      Year: { $year: "$Date" },
+      Month: { $month: "$Date" },
+      DayOfTheMonth: { $dayOfMonth: "$Date" },
+      Hour: { $hour: "$Date" },
+      Minutes: { $minute: "$Date" },
+      Seconds: { $second: "$Date" },
+      Milliseconds: { $millisecond: "$Date" },
+      Week: { $week: "$Date" },
+      DayOfTheWeek: { $dayOfWeek: "$Date" },
+      _id: 0
+    }
+  }
+]);
+```
+
+Output:
+```js
+{ 
+  "dayOfTheYear": 282, 
+  "Year": 2019, 
+  "Month": 10, 
+  "DayOfTheMonth": 9, 
+  "Hour": 7, 
+  "Minutes": 21, 
+  "Seconds": 14, 
+  "Milliseconds": 0, 
+  "Week": 40, 
+  "DayOfTheWeek": 4 
+}
+```
+
+---
+## Practice
+
+Fill in the blanks below to get all parts of the date the document with `name: "Bulbasaur"` was created on.
+```js
+db.pokemon.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      dayOfTheYear: { ???: "$Date" },
+      Year: { ???: "$Date" },
+      Month: { ???: "$Date" },
+      DayOfTheMonth: { ???: "$Date" },
+      Hour: { ???: "$Date" },
+      Minutes: { ???: "$Date" },
+      Seconds: { ???: "$Date" },
+      Milliseconds: { ???: "$Date" },
+      Week: { ???: "$Date" },
+      DayOfTheWeek: { ???: "$Date" },
+      _id: 0
+    }
+  }
+]);
+```
+
+* `$dayOfYear`
+* `$year`
+* `$month`
+* `$dayOfMonth`
+* `$hour`
+* `$minute`
+* `$second`
+* `$millisecond`
+* `$week`
+* `$dayOfWeek`
+* `$dayOfTheYear`
+* `$dayOfTheMonth`
+* `$dayOfTheWeek`
+* `$Year`
+* `$Hour`
+* `$Second`
+* `$Minute`
+* `$Week`
+* `$Month`
```

Comment:
I think this question is still too easy, with the answers already being given in the question text. also, there are too many possible answers. can you make the question ask something like "Extract the day and the hour from the given date"

Reply:
Well, all the other insights ask for each part of the date. This is why this one, asks for all of them. We can add this question as a single RQ for the whole workout and leave this insight questionless?

Reply:
hmm, yeah, you're right. let's just leave it as is.